internal/controller/utils.go

```diff
@@ -302,3 +302,25 @@ func verifySelectorObject(selector acpv1.Selector) error {
 
 	return nil
 }
+
+func shouldCreateOrUpdate(existingSecretReferences map[string]*loader.TargetSecretReference, secretName string, secretReference *loader.TargetSecretReference) bool {
```

Comment:
Why not pass the existing and target secretMetadata as input?

Reply:
Target secretMetadata may not exist in existingSecretReferences map